src/Raven.Server/Web/Studio/UpgradeInfoHandler.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Raven.Server.Commercial;
+using Raven.Server.Json;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Web.Studio;
+
+public sealed class UpgradeInfoHandler : ServerRequestHandler
+{
+    [RavenAction("/studio/upgrade-info", "GET", AuthorizationStatus.ValidUser, EndpointType.Read)]
+    public async Task GetUpgradeInfo()
+    {
+        if (ServerVersion.Build == ServerVersion.DevBuildNumber)
+        {
+            HttpContext.Response.StatusCode = (int)HttpStatusCode.NoContent;
+            return;
+        }
+            
+        var licenseId = ServerStore.LoadLicense()?.Id.ToString();
+        
+        var request = new UpgradeInfoRequest() { CurrentFullVersion = ServerVersion.FullVersion, LicenseId = licenseId };
+
+        var requestPayload = JsonConvert.SerializeObject(request);
+
+        var content = new StringContent(requestPayload, Encoding.UTF8, "application/json");
+        
+        try
+        {
+            var upgradeInfoResponse = await ApiHttpClient.Instance.PostAsync("api/v1/upgrade/info", content);
```

Comment:
Is it really `v1`? You have added it recently

Reply:
It's version number of the controller handling this request